### PR TITLE
Feat: Search Bar

### PR DESCRIPTION
**Pull request by ParthJohri**
# Feat: Search Bar [Issue No: 2](https://github.com/monacodelisa/LGBTQ-definitions/issues/2)

## Changes

1. Added the search bar, used to search the starting of the term to filter the LGBTQ terms.
2. Added necessary css for the mobile version and desktop version.
3. Added Not Found gif, if no terms matches with the search field.
4. Updated the home component html to include the search bar.

## What type of PR is this? 

- [x] ✨ Feature

## Mobile & Desktop Screenshots/Recordings

https://github.com/monacodelisa/LGBTQ-definitions/assets/76129377/e7f29c4b-2ae2-4571-b284-164df87bebb2

### CODE REVIEW

File: public/images/error404.gif
Comment (monacodelisa) [inline public/images/error404.gif:1]: error 404 is when a requested resource is not found, which is more like trying to access a page that doesn't exist
when it comes to searching a much more suitable messages could be "no results" or "no matches found"

File: src/app/components/home/home.component.html [hunk 1]
```diff
@@ -1,22 +1,31 @@
+<form [formGroup]="filterForm" class="search-form">
+  <span class="p-input-icon-left">
```
Comment (monacodelisa) [inline src/app/components/home/home.component.html:2]: the search bar is in the sidenav there is no need to create a new search bar

File: src/app/components/home/home.component.ts [hunk 2]
```diff
@@ -1,26 +1,61 @@
...
 export class HomeComponent implements OnInit {
   dataService = inject(DataService);
   LGBTQTerms: LGBTQTerm[] = [];
+  filteredTerms: LGBTQTerm[] = [];
+  filterForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.filterForm = this.formBuilder.group({
+      filter: [''],
+    });
```
Comment (monacodelisa) [inline src/app/components/home/home.component.ts:26]: forms are more suitable when we want to collect some data from the user
in this case we are not collecting any data

File: src/app/components/home/home.component.ts [hunk 3]
```diff
@@ -1,26 +1,61 @@
...
+    return this.filteredTerms
+  }
+
+  clearFilter() {
+    this.filteredTerms = [...this.LGBTQTerms];
+    this.filterForm.get('filter')?.setValue('');
+  }
+
+  trackByTerm(index: number, term: LGBTQTerm): string {
+    return term.name;
```
Comment (monacodelisa) [inline src/app/components/home/home.component.ts:55]: the inclusion of terms and types is a nice feature

**Review by monacodelisa — CHANGES_REQUESTED** [review]
The way I see it the simplest and most straightforward way to handle the search is to create a duplicate array where only the matched values are pushed and later on displayed.

The tricky part here is that the search term value comes from one component and it needs to be passed along to another component.

and there are few options of handling that, I am probably going to adjust the structure a bit so the communication would be from parent component to a child component because I would prefer not to involve a service for this, there is no need for it.

In terms of your implementation - you have implemented nice things and they can definitely be used in different situations.

My two main advices would be: 
- always use what is available first, have a good look around and if something is already provided use it.
- slow down a bit, it feels like you are rushing trying to implement things at bigger jumps while it is far better if you do things gradually. 

I have several open source projects have a look at all of them and do smaller things first before going further to things that are unclear.


And please do not erase the PR template, it is specified in the documentation that erasing the templates should not be done, you can add to them but not replacing them, specifically the part `closes#` must remain because this will link and close the tagged issue automatically upon successful merge.

Including a recording was cool I liked that 

I am closing this PR because I need to do a structural change first

**Comment by ParthJohri** [comment]
Thank you @monacodelisa for your suggestions, sure would look at other open source repos to get better understanding.
Regarding the parent to child component, I actually created a search bar component and passed the data from search bar to the home component, but I actually thought that it might create another set of component, so went on to use the logic inside the home component. Sadly, I did not see the sidebar searchbar, due to which created another component.

File: public/images/error404.gif
Comment (ParthJohri) [reply public/images/error404.gif:1]: Ya, actually I was stumbling across a number of gifs for no results, and this one gif looked cool so added it, due to which  I did overlooked the 404 error fact :' )

**Pull request closed without merging**